src/goto-programs/goto_program.h

```diff
@@ -424,42 +420,6 @@ class goto_programt
     return t;
   }
 
-  /// Get the id of the function that contains the instruction pointed-to by the
-  /// given instruction iterator.
-  ///
-  /// \param l: instruction iterator
-  /// \return id of the function that contains the pointed-to goto instruction
-  static const irep_idt get_function_id(
-    const_targett l)
-  {
-    // The field `function` of an instruction may not always contain the id of
-    // the function it is currently in, due to goto program modifications such
-    // as inlining. For example, if an instruction in a function `f` is inlined
-    // into a function `g`, the instruction may, depending on the arguments to
-    // the inliner, retain the original value of `f` in the function field.
-    // However, instructions of type END_FUNCTION are never inlined into other
-    // functions, hence they contain the id of the function they are in. Thus,
-    // this function takes the END_FUNCTION instruction of the goto program and
-    // returns the value of its function field.
-
-    while(!l->is_end_function())
-      ++l;
-
-    return l->function;
-  }
-
-  /// Get the id of the function that contains the given goto program.
-  ///
-  /// \param p: the goto program
-  /// \return id of the function that contains the goto program
-  static const irep_idt get_function_id(
```

Comment:
This function is actually used and thus any of its call-sites need to be modified.